### PR TITLE
Add a `resolve` module "hook" in BM Flow

### PR DESCRIPTION
**Pull request by amitdahan**
### 🔦 Summary
Adds a `resolve` module hook (similar to #2679) that allows firing a promise in parallel to lazy component load.

By having the following in `src/pages/foo.module.ts`:

```typescript
import { ResolveFn } from 'yoshi-flow-bm-runtime';

export const resolve: ResolveFn = async () => ({
  foo: await fetchFoo(),
});
```

Our component can be injected with the `resolve`d promise in the form of props:
```typescript jsx
// src/pages/foo.tsx
const Foo: FC<FooProps> = ({ foo }) => {
  return <pre>{foo}</pre>;
};

export default Foo;
```

### CODE REVIEW

File: website/versioned_docs/version-4.x/business-manager-flow/overview.md [hunk 1]
```diff
@@ -62,6 +62,25 @@ export const files: FilesFn = ({ moduleParams }) => [
 ]
 ```
 
+#### Resolving data during load
```
Comment (ranyitz) [inline website/versioned_docs/version-4.x/business-manager-flow/overview.md:65]:
```suggestion
#### Resolving data parallel to page load
```

File: website/versioned_docs/version-4.x/business-manager-flow/overview.md [hunk 2]
```diff
@@ -88,6 +107,25 @@ export const files: FilesFn = ({ moduleParams }) => [
...
```
Comment (ranyitz) [inline website/versioned_docs/version-4.x/business-manager-flow/overview.md:110]:
```suggestion
#### Resolving parallel to component load
```

File: website/versioned_docs/version-4.x/business-manager-flow/overview.md [hunk 3]
```diff
@@ -88,6 +107,25 @@ export const files: FilesFn = ({ moduleParams }) => [
...
+   
+Create a file ending with `.module.ts` (if it doesn't exist) next to your existing page.
```
Comment (ranyitz) [inline website/versioned_docs/version-4.x/business-manager-flow/overview.md:112]:
```suggestion
Create a file ending with `.module.ts` (if it doesn't exist) next to your existing component.
```

Comment (amitdahan) [reply website/versioned_docs/version-4.x/business-manager-flow/overview.md:112]: Thanks :)

**Pull request merged**